[jit][edge] Implement torch::jit::Function for mobile funciton.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65972
* #65971
* __->__ #65970
* #65969





mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

zhxchen17 committed:
[jit][edge] Implement torch::jit::Function with an Executor for mobile funciton.



Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31326148/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function with an Executor for mobile funciton."


Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31326148/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM. Please make sure to run internal BSB tool, and record the size results in this PR, before it's landed.

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Implement torch::jit::Function for mobile funciton."






mobile::Function should inherit from jit::Function, because for interface call support, we need an abstract jit::Function type stored in corresponding ClassTypes, so that we can look up methods in there. Previously mobile::Function is implemented separately which prevents this. Since we get rid of all the unneeded virtual methods from jit::Function, we can inherit from torch::jit::Function without too much cost.

NOTE that torch::jit::Function is already in dependency because we need it to support custom class call. We should be able to use Function uniformly without looking into whether it's a builtin function or mobile::Function.

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148/)

Differential Revision: [D31326148](https://our.internmc.facebook.com/intern/diff/D31326148)

[ghstack-poisoned]
